src/dotnet/APIView/APIViewWeb/wwwroot/js/site.js

```diff
@@ -4,10 +4,54 @@
 // Write your Javascript code.
 $(function () {
     $(".commentable").click(function () {
-        var myForm = $("#commentForm");
-        $("#idBox").val(this.id);
-        myForm.show();
-        $("#commentThread").focus();
+        var nextRow = $(this).parents(".code-line").first().next();
+        var formExists = nextRow.find(".comment-form").length > 0;
+
+        if (!formExists) {
+            let myForm = $(".comment-form");
+            let clone = myForm.clone();
+            clone.first().find(".id-box").val(this.id);
+            let stringRep = clone[0].outerHTML;
+
+            var thread = nextRow.find(".comment-thread-contents");
+            if (thread.length > 0) {
+                thread.after(stringRep);
+                nextRow.find(".review-thread-reply").hide();
+            }
+            else {
+                $(this).parents(".code-line").after("<tr><td>" + stringRep + "</td></tr>");
+                nextRow = $(this).parents(".code-line").first().next();
+            }
+        }
+        nextRow.find(".comment").show();
+        nextRow.find(".new-thread-comment-text").focus();
         return false;
     });
 });
+
+$(function () {
+    $(".review-thread-reply-button").click(function () {
```

Comment:
Extract whatever click handler does into a method and call it from here, don't fire events on other elements to initiate the behavior.